server/src/main/java/org/elasticsearch/gateway/AsyncShardFetch.java

```diff
@@ -57,6 +58,7 @@
     private final Set<String> nodesToIgnore = new HashSet<>();
     private final AtomicLong round = new AtomicLong();
     private boolean closed;
+    private final AtomicInteger fetchingCount = new AtomicInteger();
```

Comment:
I think this could just be a `volatile int` because it's only ever updated within `synchronized` methods.